### PR TITLE
Parallel process replay

### PR DESCRIPTION
**Pull request by lukaspetersson**
Use multiprocessing to speed up process replay.

Branched from PRs: https://github.com/commaai/openpilot/pull/24495 and https://github.com/commaai/openpilot/pull/24496
Uses Cereal from PR: https://github.com/commaai/cereal/pull/299

### CODE REVIEW

File: selfdrive/common/params.cc [hunk 1]
```diff
@@ -180,9 +184,30 @@ std::unordered_map<std::string, uint32_t> keys = {
 
 } // namespace
 
+void clean_param_dirs(std::string path){
```
Comment (adeebshihadeh) [inline selfdrive/common/params.cc:187]: why clean the directories? I don't think it's params' responsibility to do this

File: selfdrive/common/params.cc [hunk 2]
```diff
@@ -180,9 +184,30 @@ std::unordered_map<std::string, uint32_t> keys = {
...
+    LOGW("File already deleted in a parallel process");
+  }
+}
+
 Params::Params(const std::string &path) {
   static std::string default_param_path = ensure_params_path();
   params_path = path.empty() ? default_param_path : ensure_params_path(path);
+  prefix = std::getenv("OPENPILOIT_PREFIX");
+  clean_param_dirs(getParamPath());
+  std::filesystem::create_directory(getParamPath());
```
Comment (adeebshihadeh) [inline selfdrive/common/params.cc:210]: `create_params_path ` already takes care of this

File: selfdrive/test/process_replay/test_processes.py [hunk 3]
```diff
@@ -154,36 +181,43 @@ def format_diff(results, ref_commit):
...
-    results[segment] = {}
-
-    r, n = segment.rsplit("--", 1)
-    lr = LogReader(get_url(r, n))
-
+    if not args.parallel:
+      print(f"***** testing route segment {segment} *****\n")
+      results[segment] = {}
+      r, n = segment.rsplit("--", 1)
+      lr = LogReader(get_url(r, n))
```
Comment (adeebshihadeh) [inline selfdrive/test/process_replay/test_processes.py:193]: this shouldn't be a completely separate code path. process replay should just take something like a jobs argument

File: selfdrive/test/process_replay/test_processes.py [hunk 4]
```diff
@@ -127,6 +152,8 @@ def format_diff(results, ref_commit):
                       help="Updates reference logs using current commit")
   parser.add_argument("--upload-only", action="store_true",
                       help="Skips testing processes and uploads logs from previous test run")
+  parser.add_argument("--parallel", action="store_true",
+                      help="Run tests in parallel")
```
Comment (adeebshihadeh) [inline selfdrive/test/process_replay/test_processes.py:156]:
```suggestion
  parser.add_argument("--jobs", action="store_true")
```

File: selfdrive/test/process_replay/test_processes.py (hunk 3 above)
Comment (lukaspetersson) [reply selfdrive/test/process_replay/test_processes.py]: I did it that way initially, but there is then a lot of duplicate code. I will change it back and we can see

**Pull request closed without merging**